src/NumericalAlgorithms/SphericalHarmonics/RealSphericalHarmonics.cpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/SphericalHarmonics/RealSphericalHarmonics.hpp"
+
+#include <boost/math/special_functions/spherical_harmonic.hpp>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+void real_spherical_harmonic(gsl::not_null<DataVector*> spherical_harmonic,
+                             const DataVector& theta, const DataVector& phi,
+                             size_t l, int m) {
+  ASSERT(size_t(abs(m)) <= l, "m needs to be smaller than l");
+  for (size_t i = 0; i < theta.size(); ++i) {
+    if (m < 0) {
+      const int sign = (m % 2 == 0) ? 1 : -1;
+      spherical_harmonic->at(i) =
```

Comment:
I actually didn't know `DataVector` had an `at` function. Maybe switch to `(*spherical_harmonic)[i]` to be consistent with other parts of the code? At least `std::vector` the difference between `[]` and `at` is that `at` _always_ incurs the overhead of a bounds check. For `DataVector`, `[]` only bounds checks in debug mode

Same for `theta` and `phi`